src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/MessageDrivenSubscriptionsConfigExtensions.cs

```diff
@@ -42,23 +42,43 @@ public static class MessageDrivenSubscriptionsConfigExtensions
         /// Registers a publisher endpoint for a given event type.
         /// </summary>
         /// <param name="routingSettings">The <see cref="RoutingSettings&lt;T&gt;" /> to extend.</param>
-        /// <param name="publisherEndpoint">The publisher endpoint.</param>
         /// <param name="eventType">The event type.</param>
-        public static void RegisterPublisherForType<T>(this RoutingSettings<T> routingSettings, string publisherEndpoint, Type eventType) where T : TransportDefinition, IMessageDrivenSubscriptionTransport
+        /// <param name="publisherEndpoint">The publisher endpoint.</param>
+        public static void RegisterPublisherForType<T>(this RoutingSettings<T> routingSettings, Type eventType, string publisherEndpoint) where T : TransportDefinition, IMessageDrivenSubscriptionTransport
         {
-            routingSettings.Settings.GetOrCreate<Publishers>().Add(publisherEndpoint, eventType);
+            Guard.AgainstNullAndEmpty(nameof(publisherEndpoint), publisherEndpoint);
+
+            routingSettings.Settings.GetOrCreate<Publishers>().Add(eventType, publisherEndpoint);
         }
 
         /// <summary>
-        /// Registers a publisher endpoint for all event types in a given assembly and, optionally, namespace.
+        /// Registers a publisher endpoint for all event types in a given assembly.
         /// </summary>
         /// <param name="routingSettings">The <see cref="RoutingSettings&lt;T&gt;" /> to extend.</param>
+        /// <param name="eventAssembly">The assembly containing the event types.</param>
         /// <param name="publisherEndpoint">The publisher endpoint.</param>
+        public static void RegisterPublisherForAssembly<T>(this RoutingSettings<T> routingSettings, Assembly eventAssembly, string publisherEndpoint) where T : TransportDefinition, IMessageDrivenSubscriptionTransport
+        {
+            Guard.AgainstNull(nameof(eventAssembly), eventAssembly);
+            Guard.AgainstNullAndEmpty(nameof(publisherEndpoint), publisherEndpoint);
+
+            routingSettings.Settings.GetOrCreate<Publishers>().Add(eventAssembly, publisherEndpoint);
+        }
+
+        /// <summary>
+        /// Registers a publisher endpoint for all event types in a given assembly and namespace.
+        /// </summary>
+        /// <param name="routingSettings">The <see cref="RoutingSettings&lt;T&gt;" /> to extend.</param>
         /// <param name="eventAssembly">The assembly containing the event types.</param>
         /// <param name="eventNamespace">The namespace containing the event types.</param>
```

Comment:
MessageMapping not. Assembly scanning has pattern march support. Not saying
we should have it. Just thought if we need to mention that matching is not
supported?

On Wednesday, July 6, 2016, Tim Bussmann <notifications@github.com> wrote:

> In
> src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/MessageDrivenSubscriptionsConfigExtensions.cs
> <https://github.com/Particular/NServiceBus/pull/3898#discussion_r69757025>
> :
>
> >          /// <param name="publisherEndpoint">The publisher endpoint.</param>
> > +        public static void RegisterPublisherForAssembly<T>(this RoutingSettings<T> routingSettings, Assembly eventAssembly, string publisherEndpoint) where T : TransportDefinition, IMessageDrivenSubscriptionTransport
> > +        {
> > +            Guard.AgainstNull(nameof(eventAssembly), eventAssembly);
> > +            Guard.AgainstNullAndEmpty(nameof(publisherEndpoint), publisherEndpoint);
> > +
> > +            routingSettings.Settings.GetOrCreate<Publishers>().Add(eventAssembly, publisherEndpoint);
> > +        }
> > +
> > +        /// <summary>
> > +        /// Registers a publisher endpoint for all event types in a given assembly and namespace.
> > +        /// </summary>
> > +        /// <param name="routingSettings">The <see cref="RoutingSettings&lt;T&gt;" /> to extend.</param>
> >          /// <param name="eventAssembly">The assembly containing the event types.</param>
> >          /// <param name="eventNamespace">The namespace containing the event types.</param>
>
> do we support wild cards on one of our APIs?
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Particular/NServiceBus/pull/3898/files/dfc1ba59b104f467087ace0487c883e730be1548#r69757025>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAKosqDfFyXkbMsRnkQGpRmWXr6Z8n5uks5qS84xgaJpZM4JC_FI>
> .
>